go/cli/commands/build.go

```diff
@@ -30,80 +30,121 @@ import (
 
 //go:embed embed/Dockerfile
 var f embed.FS
+var execCmdFn = execCmd
+var execLookPathFn = exec.LookPath
 
 const (
-	ImageTag              = "function:latest"
+	// Builder Type
+	Ko     = "ko"
+	Docker = "docker"
+
+	// Docker constant variables
+	Image                 = "function:latest"
 	DockerfilePath        = "Dockerfile"
-	builtinDockerfilePath = "embed/Dockerfile"
+	BuiltinDockerfilePath = "embed/Dockerfile"
+
+	// Ko constant variables
+	KoDockerRepoEnvVar = "KO_DOCKER_REPO"
+	KoLocalRepo        = "ko.local"
 )
 
-func NewBuildCmd(ctx context.Context) *cobra.Command {
+func NewBuildRunner(ctx context.Context) *BuildRunner {
 	r := &BuildRunner{
-		ctx: ctx,
+		ctx:    ctx,
+		Ko:     &KoBuilder{},
+		Docker: &DockerBuilder{},
 	}
 	r.Command = &cobra.Command{
-		Use:     "build",
-		Short:   "build the KRM function as a docker image",
-		PreRunE: r.PreRunE,
-		RunE:    r.RunE,
-	}
-	r.Command.Flags().StringVarP(&r.Tag, "tag", "t", ImageTag,
-		"the docker image tag")
-	r.Command.Flags().StringVarP(&r.DockerfilePath, "file", "f", "",
-		"Name of the Dockerfile. If not given, using a default builtin Dockerfile")
-	return r.Command
+		Use:   "build",
+		Short: "build your KRM function to a container image",
+		RunE:  r.RunE,
+	}
+	r.Command.Flags().StringVarP(&r.BuilderType, "builder", "b", Ko,
+		"the image builder. `ko` is the default builder, which requires `go build`; `docker` is accepted, and "+
+			" requires you to have docker installed and running")
+	r.Command.Flags().StringVarP(&r.Docker.Image, "image", "i", Image,
```

Comment:
Yeah, I had a todo to better align the `-image` and `-tag` flags. It would require several corner case handling which are not quite the focus of this PR.